esm_parser/esm_parser.py

```diff
@@ -238,6 +246,12 @@ def initialize_from_yaml(filepath):
     for file_ending in YAML_AUTO_EXTENSIONS:
         if filepath.endswith(file_ending) and not file_ending == "":
             user_config = yaml_file_to_dict(filepath)
+            
+            if not "general" in user_config:
+                user_config['general'] = {}
+            # full absolute path of the user runscript including the yaml file
+            user_config['general']['runscript_abspath'] = filepath
```

Comment:
Nothing to change, I'm just curious to know in which scenario would the `general` section not exist?

Reply:
I asked myself the same question 😄 , I decided to add this after seeing 

```python
def complete_config(user_config):
    if not "general" in user_config:
        user_config["general"] = {}
    user_config["general"]["additional_files"] = []
```

since `complete_config` is called at the end of the `initialize_from_yaml()`, I decided to check it too. Better be safe than sorry 😄

Reply:
👍